src/transcript-plugin.tsx

```diff
@@ -1,458 +1,306 @@
-      label: "Transcript",
-      expandMode: expandMode === KitchenSinkExpandModes.OverTheVideo ?
-        KitchenSinkExpandModes.OverTheVideo :
-        KitchenSinkExpandModes.AlongSideTheVideo,
-      renderIcon: () => (
-        <button
-          className={styles.pluginButton}
-          tabIndex={1}
-          onClick={this._handleIconClick}
-        >
-        <div className={styles.transcriptIcon} />
-        </button>
-      ),
-      position: getConfigValue(
-          position,
-          position =>
-              typeof position === "string" &&
-              (position === KitchenSinkPositions.Bottom ||
-                  position === KitchenSinkPositions.Right),
-          KitchenSinkPositions.Bottom
-      ),
-      renderContent: this._renderKitchenSinkContent
-    });
+  onRegisterUI(): void {}
 
-    if (expandOnFirstPlay) {
-      this._kitchenSinkItem.activate();
+  private _handleLanguageChange = () => {
+    this._activeCaptionMapId = this._getCaptionMapId();
+    if (this._captionMap.has(this._activeCaptionMapId)) {
+      this._updateTranscriptPanel();
     }
-  }
-
-  private _onTimeUpdate = (): void => {
-    this._updateKitchenSink();
   };
 
-  private _updateKitchenSink() {
-    if (this._kitchenSinkItem) {
-      this._kitchenSinkItem.update();
-    }
-  }
-
-  private _initLoading() {
-    if (!this._isLoading || this._hasError) {
-      this._isLoading = true;
-      this._hasError = false;
-      this._updateKitchenSink();
+  private _updateTranscriptPanel() {
+    this._isLoading = false;
+    if (this._transcriptPanel) {
+      this.sidePanelsManager.update(this._transcriptPanel);
     }
   }
 
-  private _onError = (error?: Error, message?: string, method?: string) => {
-    let msg: string = message || "Error message not defined";
-    if (error instanceof KalturaClientException) {
-      msg = `${msg} (network error etc)`;
-    } else if (error instanceof KalturaAPIException) {
-      msg = `${msg} (api exception)`;
-    }
-    logger.error(msg, {
-      method: method || "Method not defined",
-      data: {
-        error
+  private _onTimedMetadataAdded = ({payload}: TimedMetadataEvent) => {
+    const captionData: ItemData[] = [];
+    payload.cues.forEach((cue: CuePoint) => {
+      if (this.isCaptionType(cue)) {
+        captionData.push(prepareCuePoint(cue, ItemTypes.Caption));
       }
     });
-    this._isLoading = false;
-    this._hasError = true;
-    this._updateKitchenSink();
+    if (captionData.length) {
+      this._addCaptionData(captionData);
+    }
   };
 
-  private _loadCaptions = (event?: {}): void => {
-    if (!this._entryId) {
-      return;
+  private _onTimedMetadataChange = ({payload}: TimedMetadataEvent) => {
+    const transcriptCuePoints: Array<CuePoint> = payload.cues;
+    this._activeCuePointsMap = {};
+    if (transcriptCuePoints.length) {
+      const latestTranscriptCuePoint = transcriptCuePoints[transcriptCuePoints.length - 1];
```

Comment:
we don't need to group cue-points
Change `this._activeCuePointsMap` according to timed metadata